src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -705,10 +705,10 @@ private async ValueTask<TResult> ReceiveAsyncPrivate<TResult>(Memory<byte> paylo
             try
             {
                 await _receiveMutex.EnterAsync(cancellationToken).ConfigureAwait(false);
-                ObjectDisposedException.ThrowIf(_disposed, typeof(WebSocket));
-
                 try
                 {
+                    ObjectDisposedException.ThrowIf(_disposed, typeof(WebSocket));
```

Comment:
If I am reading this correctly, this exception will end up in the `catch` block and exception of a different type will be thrown instead.

Reply:
Nevermind, misread the indentation on GitHub.